### PR TITLE
Allow for anonymous commenting.

### PR DESCRIPTION
**Pull request by smn**

### CODE REVIEW

**Comment by smn** [comment]
this_is_an_article_with_no_image_hero_image_uploaded

Now allowing anonymous comments.

**Comment by smn** [comment]
127_0_0_1_8000_comments_molo_post_

Also we have a page for when the form doesn't validate. It inherits the styling from the page the comments were originally linked to.

I've not applied much styling of the individual error.

**Comment by smn** [comment]
this_is_an_article_with_no_image_hero_image_uploaded

This is what's presented when someone's not logged in. Needs proper styling still as well.

**Comment by smn** [comment]
@miltontony, @bruskiza & @saeedmb8 ready for review next week.

File: tuneme/templates/comments/form.html [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+    {% if form.errors %}
+      <h1>{% blocktrans count counter=form.errors|length %}Please correct the error below{% plural %}Please correct the
+        errors below{% endblocktrans %}</h1>
+    {% endif %}
 
+    {% if form.comment.errors %}{{ form.comment.errors }}{% endif %}
     <p>{{ form.comment }}</p>
     <input type="submit" name="submit" value="Comment">
-    {# <input type="submit" name="submit_anonymously" value="Add anonymously">#}
+    <input type="submit" name="submit_anonymously" value="Add anonymously">
```
Comment (smn) [inline tuneme/templates/comments/form.html:14]: Not this is a "magic" flag recognised by molo.commenting, if that's present as a key in `request.POST` it'll use `Anonymous` as the name of the person submitting while still linking it to the user's profile.

Future releases should make `Anonymous` as a bit of text configurable.

Comment (smn) [inline tuneme/templates/comments/form.html:14]: https://github.com/praekelt/molo.commenting/issues/4 is the issue for that.

File: tuneme/templates/comments/preview.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      {% if self.image %}
+        {% image self.image width-98 %}
+      {% else %}
+        {% image self.get_parent.image width-98 %}<!-- ??????? -->
+      {% endif %}
+  </div>
+     <h6>{{get_parent_section.title}}</h6><!-- ??????? -->
+    {%  with parent=self.get_ancestors.last %}
+      <h6>{{parent.title}}</h6>
+    {% endwith %}
```
Comment (smn) [inline tuneme/templates/comments/preview.html]: @saeedmb8 I'll be adding the fixes you made to the images here as well.

File: tuneme/templates/comments/preview.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+</div>
+
+<div class="block {{self.get_parent_section.get_effective_extra_style_hints}}">
+  {% if messages %}
+  <ul class="messages">
+      {% for message in messages %}
+      <li{% if message.tags %} class="{{ message.tags }}"{% endif %}>{{ message }}</li>
+      {% endfor %}
+  </ul>
```
Comment (smn) [inline tuneme/templates/comments/preview.html:30]: We don't have any styling for messages yet.

**Comment by smn** [comment]
Ready for review, would like to land this soon so we can make it part of the production deploy.

**Comment by smarzban** [comment]
:+1:

**Pull request merged**